File: HerStory.Server/Models/Portrait.cs
using System.Text.Json.Serialization;

namespace HerStory.Server.Models
{
    public class Portrait
    {
        public int Id { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public required DateTime DateOfBirth { get; set; }
        public DateTime? DateOfDeath { get; set; }
        public required string BiographyAbstract { get; set; }
        public required string BiographyFull { get; set; }
        public string? PhotoUrl { get; set; }
        public required string CountryOfBirth { get; set; }
        public required DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public ICollection<PortraitCategory> PortraitCategories { get; set; } = new List<PortraitCategory>();
        public  ICollection<PortraitField> PortraitFields { get; set; } = new List<PortraitField>();
    }
}

File: herstory.client/src/app/interfaces/portrait.ts
export interface PortraitCard {
  id: number;
  firstName: string;
  lastName: string;
  dateOfBirth: Date;
  dateOfDeath: Date;
  photoUrl: string;
  biographyAbstract: string;
  categories: Category[];
  fields : Field[];
}

export interface PortraitDetail {
  id: number;
  firstName: string;
  lastName: string;
  dateOfBirth: Date;
  dateOfDeath: Date;
  photoUrl: string;
  biographyAbstract: string;
  biographyFull: string;
  countryOfBirth: string;
  categories: Category[];
  fields: Field[];
}

export interface Category {
  id: number;
  name: string;
}

export interface Field {
  id: number;
  name: string;
}

export interface FilterCriteria {
  categories: Category[];
  fields: Field[];
}

